File: BE/src/main/java/com/ssafy/tax7i/card/dto/CardProductResponse.java
package com.ssafy.tax7i.card.dto;

import com.ssafy.tax7i.banking.client.dto.SsafyCreditCardProductListResponse;

public record CardProductResponse(
        String cardUniqueNo,
        String cardIssuerName,
        String cardName,
        String baselinePerformance,
        String maxBenefitLimit,
        String cardDescription
) {
    public static CardProductResponse from(SsafyCreditCardProductListResponse.CreditCardProductRec rec) {
        return new CardProductResponse(
                rec.cardUniqueNo(),
                rec.cardIssuerName(),
                rec.cardName(),
                rec.baselinePerformance(),
                rec.maxBenefitLimit(),
                rec.cardDescription()
        );
    }
}

File: FE/app/src/main/java/com/ssafy/seveniTax/data/model/card/CardProductResponse.kt
package com.ssafy.seveniTax.data.model.card

data class CardProductResponse(
    val cardUniqueNo: String,
    val cardIssuerName: String,
    val cardName: String,
    val baselinePerformance: String,
    val maxBenefitLimit: String,
    val cardDescription: String
)
